test/mocha/20-tokens.js

```diff
@@ -1,18 +1,398 @@
+      const internalId = 'aM6pup9s6XTaYTho';
+      const requester = 'requester';
+      let err;
+      let result;
+      try {
+        const {tokens: tks} = await tokens.create(
+          {internalId, attributes, tokenCount});
+        const token = tks[0];
+        // intentionally clear tokenBatch database to remove token created
+        await cleanBatchDB();
+        result = await tokens.resolve({requester, token});
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(result);
+      should.exist(err);
+      err.name.should.equal('NotFoundError');
+      err.message.should.equal('Token not found.');
+    });
+  it('should resolve token to the party identified by "requester"',
+    async function() {
+      const tokenCount = 1;
+      const internalId = 'aM6pup9s6XTaYTho';
+      const attributes = Uint8Array.from(new Set([1]));
+      const requester = 'requester';
+      let err;
+      let result;
+      let tks;
+      try {
+        tks = await tokens.create(
+          {internalId, attributes, tokenCount});
+        const token = tks.tokens[0];
+        result = await tokens.resolve({requester, token});
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(err);
+      areTokens(tks);
+      should.exist(result.pairwiseToken);
+    });
+  it('should resolve token when called twice with same "requester"',
+    async function() {
+      const tokenCount = 1;
+      const internalId = 'aM6pup9s6XTaYTho';
+      const attributes = Uint8Array.from(new Set([1]));
+      const requester = 'requester';
+      let err;
+      let result1;
+      let result2;
+      let tks;
+      try {
+        tks = await tokens.create(
+          {internalId, attributes, tokenCount});
+        const token = tks.tokens[0];
+        result1 = await tokens.resolve({requester, token});
+        // resolve token with same requester again
+        result2 = await tokens.resolve({requester, token});
+
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(err);
+      areTokens(tks);
+      should.exist(result1.pairwiseToken);
+      should.exist(result2.pairwiseToken);
+      result1.pairwiseToken.should.equal(result2.pairwiseToken);
+    });
+  it('should throw error when token is resolved with different "requester"',
+    async function() {
+      const tokenCount = 1;
+      const internalId = 'aM6pup9s6XTaYTho';
+      const attributes = Uint8Array.from(new Set([1]));
+      const requester1 = 'requester1';
+      const requester2 = 'requester2';
+      let err;
+      let result1;
+      let result2;
+      let tks;
+      try {
+        tks = await tokens.create(
+          {internalId, attributes, tokenCount});
+        const token = tks.tokens[0];
+        result1 = await tokens.resolve({requester: requester1, token});
+        // resolve token with different requester
+        result2 = await tokens.resolve({requester: requester2, token});
+
+      } catch(e) {
```

Comment:
This is not a good pattern since we don't know which of the requests is failing.  There are three requests here that might potentially throw an error.  The test is only expecting one of them to throw.  Maybe it's not necessary to have the first request that is expected to succeed in a try catch at all.

This applies to any other place here where you're wrapping more than one command in a try catch.